### ISSUE TITLE
Add transactions with amounts 100 and 10.50=> splits table shows 100 and 21 => Trial Balance shows 121

### ISSUE THREAD

**Issue by rlyders** [darcys22/godbledger#86]
version: godbledger-linux-x64-v0.4.0
Windows 10 OS Build 19041.508
Windows Subsystem for Linux 2 (WSL 2)

Thanks for the work to produce godbledger. Lots of good stuff you guys have created. Please see my issue below and let me know if I am missing something. Thanks for your help.

ISSUE:
There appears to be an issue handling the conversion of decimal values when these values are shown in the trial balance. When I create 2 new transactions with amounts 100 and 10.50 those amounts result in a value of 121 in the displayed trial balance. I understand that the splits table converts the 10.50 to 21. I just don't see where it is supposed to be reverting it back to 10.50 for display purposes.

STEPS TO REPRODUCE:
Per quick start: https://github.com/darcys22/godbledger/wiki/Quickstart

Clear database in one shell:
```
rm -fr ~/.ledger && ~/godbledger/godbledger init && ~/godbledger/godbledge
```

Add 2 new records in 2nd shell. The ledger_cli response shows that 10.5 was converted to 21/2.
```
$ ~/godbledger/ledger_cli jsonjournal '{"Payee":"Monthly Savings","Date":"2020-09-28T00:00:00Z","Description":"Boxes","AccountChanges":[{"Name":"Asset:Checking","Description":"DEBIT:ACCT_XFER:OUT","Currency":"USD","Balance":"-100.00"},{"Name":"Asset:Savings","Description":"DEBIT:ACCT_XFER:IN","Currency":"USD","Balance":"100.00"}]}'
[2020-09-28 13:40:49] DEBUG Config: Configuration Successfully loaded Config File=/home/rlyders/.ledger/config.toml
{Monthly Savings 2020-09-28 00:00:00 +0000 UTC [{Asset:Checking DEBIT:ACCT_XFER:OUT USD -100/1} {Asset:Savings DEBIT:ACCT_XFER:IN USD 100/1}] }
[2020-09-28 13:40:49]  INFO ledger_cli: GRPC Dialing on port address=127.0.0.1:50051
[2020-09-28 13:40:50]  INFO ledger_cli: Response: btp2scfqh58og259ljlg

$ ~/godbledger/ledger_cli jsonjournal '{"Payee":"Bill Red","Date":"2020-09-15T00:00:00Z","Description":"Chilli","AccountChanges":[{"Name":"Asset:Checking","Description":"DEBIT:BILLPAY:OUT","Currency":"USD","Balance":"-10.50"},{"Name":"Expenses","Description":"DEBIT:BILLPAY:IN","Currency":"USD","Balance":"10.50"}]}'
[2020-09-28 13:40:56] DEBUG Config: Configuration Successfully loaded Config File=/home/rlyders/.ledger/config.toml
{Bill Red 2020-09-15 00:00:00 +0000 UTC [{Asset:Checking DEBIT:BILLPAY:OUT USD -21/2} {Expenses DEBIT:BILLPAY:IN USD 21/2}] }
[2020-09-28 13:40:56]  INFO ledger_cli: GRPC Dialing on port address=127.0.0.1:50051
[2020-09-28 13:40:56]  INFO ledger_cli: Response: btp2se7qh58og259ljng
```

Show trial balance. The raw 21 amount is added to 100 without reverting the value back to 10.5
```
~/godbledger/reporter trialbalance
     ACCOUNT     | BALANCE AT 28 SEPTEMBER 2020
-----------------+-------------------------------
  Asset:Checking |                         -121
  Asset:Savings  |                          100
  Expenses       |                           21
```

Show split records in sqlite3 database
```
sqlite> select * from splits;
btp2llnqh58ohrjvq7vg|2020-09-28 00:00:00+00:00|Monthly Savings|USD|-100|btp2llnqh58ohrjvq7v0
btp2llnqh58ohrjvq800|2020-09-28 00:00:00+00:00|Monthly Savings|USD|100|btp2llnqh58ohrjvq7v0
btp2lmfqh58ohrjvq81g|2020-09-15 00:00:00+00:00|Bill Red|USD|-21|btp2lmfqh58ohrjvq810
btp2lmfqh58ohrjvq820|2020-09-15 00:00:00+00:00|Bill Red|USD|21|btp2lmfqh58ohrjvq810
```

Thanks again!
-Rich

**Comment by darcys22** [comment]
Thanks for raising this! Definitely appreciate the amount of effort it takes to create this level of detail in the bug description.

That’s definitely unintended behaviour. Pretty sure its to do with the big.Rational type within the ledger_cli client and its dropping the denominator before its even sent over wire to the godbledger server.

The intention is to keep everything as integers at this point (so $10.50 would actually be 1050) and push the processing of the decimal places higher up. So having a big.Rational here doesn't actually make sense.

I guess early on I hadn't fully decided on whether the ledger_cli client should process the decimal point or not.

But should be easy enough to to fix this one.

Thanks again!
Sean

**Comment by rlyders** [comment]
Hi Sean,
Good to hear from you so quickly. Thanks for looking into this.
When you have a fix let me know. No rush.
For now, I just truncated all my amounts so I’m ignoring the pennies. 😊

-Rich

**Comment by darcys22** [comment]
Addressed this in #87 

In the end I changed everything in ledger_cli and reporter to process decimals for the dollars rather than enforcing cents as an integer.

This will cause a breaking change in that the amounts stored in the database are as integers ($1 = 100 units) but the command line will take the whole dollar amount and multiply it by 100 before saving it to the database. 

Probably would recommend not truncating the amounts and instead storing them as cents, this pull request I've just merged will mean the next release will format everything nicely on the CLI and allow you to enter decimal places. However if you are storing the amounts as $1 = 1 then this won't sit nicely when it changes to $1 = 100
